src/main/java/AddCommand.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Represents a command to add a task to the current list of tracked tasks.
+ */
+public class AddCommand extends Command {
+    public AddCommand(String firstWord, String restOfCommand) {
+        super(firstWord, restOfCommand);
+    }
+
+    /**
+     * Determine which type of task to add before adding the right task to
+     * the array of tracked tasks.
+     * @param taskList TaskList object containing the array of tracked tasks
+     * @param ui Ui object containing methods for user interaction
+     * @param storage Storage object for dealing with saving tasks to file
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        switch (firstWord) {
+        case "todo":
+            try {
+                createTodoTask(taskList, restOfCommand);
+            } catch (DukeException e) {
+                ui.showInvalidTodoError();
+                return;
+            }
+            break;
+        case "deadline":
+            try {
+                createDeadlineTask(taskList, restOfCommand);
+            } catch (DukeException e) {
+                ui.showInvalidDeadlineError();
+                return;
+            }
+            break;
+        case "event":
+            try {
+                createEventTask(taskList, restOfCommand);
+            } catch (DukeException e) {
+                ui.showInvalidEventError();
+                return;
+            }
+            break;
+        }
+        ui.showTaskIsAdded();
+        ui.showCurrTask(taskList, taskList.numOfTasks);
+        taskList.incrementNumOfTasks();
+        ui.showCurrNumOfTask(taskList);
+    }
+
+    /**
+     * Add a new Todo task to the array of tracked tasks.
+     * @param taskList TaskList object containing the array of tracked tasks
+     * @param input A string containing data regarding the Todo task to be added
+     * @throws DukeException If input has missing information
+     */
+    public static void createTodoTask(TaskList taskList, String input) throws DukeException {
+        if ((input.trim()).length() == 0) {
+            // Task has no description
```

Comment:
This comment can be placed better maybe next to the if statement or above it rather than inside the block. It will be more intuitive that way. [See here](https://se-education.org/guides/conventions/java/basic.html#comments)